text/0023-stats-component.md

```diff
@@ -0,0 +1,345 @@
+
+# Motivation
+[motivation]: #motivation
+
+<!-- Why are we doing this? What use cases does it support? What is the expected outcome? -->
+Qri is missing statistical metadata! We have descriptive metadata (stored in the Meta component), structural metadata (Structure component), and administrative metadata (Commit component), but no statisicial metadata. This rfc proposes the the path forward to statistical metadata stored in the Stats component. 
+
+Creating and storing a Stats component will take a sliver of the computing power it takes to create and store the dataset itself (as this rfc will prove). For that small cost, the stats component will make the dataset creators life easier. It will be a quick way to learn about the full dataset body, and it will allow us to have visualizations without needing to inject the full body into the html. That not only reduces loading time for templates, but will make our dataset packages slimmer.
+
+The stats component is one part in a push to get dataset creators to an interesting, publishable insight quicker. The ideal path, using the default Stats generator and the default template would be: the creator adds a dataset with a tabular body. On dataset creation, Qri calculates a stats component, and uses this stats component to render a visualization from the default template. The creator can check out that template (or the stats component itself), to see if there are any interesting insights. The creator may choose to publish this dataset then and there, exposing the dataset and the rendered findings to the world, or create a new version of the dataset exploring the findings further.
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+Let's split this section in to three parts. 
+First we will talk about what a default stats component might looks like and what stats we want to generate. Then let's talk about future possibilities, namely what a default stats component might look like for high dimentional data and what custom stats might look like and what we need to do to get there.
+
+## Default Stats component:
+To start, Qri can create a default stats component on tabular (two dimentional) data. Because we want the cost of creating a stats component to be low, we only want to generate stats that can be calculated on a rolling basis.
+
+The different types of data that we can generate stats on are number (including integer and floating point), string, and boolean.
+
+We can generate different stats depending on the type of that column.
+
+Number:
+  valid - the number of cells in that column that contain valid numbers
+  error - the number of cells in that column that contain invalid numbers
+  missing - the number of cells in that column that do not contain data
```

Comment:
I added valid/error/missing, rather then just count after seeing in in kaggle. Feels like error is especially meaningful when we are trying to calculate stats, since values that error will not actually make into something like an average. 
We could remove missing and just use valid/error if we want to save space.